examples/campaign_management/add_complete_campaigns_using_mutate_job.py

```diff
@@ -0,0 +1,558 @@
+    ad_group_criterion.ad_group.value = ad_group_operation.create.resource_name
+    ad_group_criterion.status = client.get_type(
+        'AdGroupCriterionStatusEnum', version='v3').ENABLED
+
+    return ad_group_criterion_operation
+
+
+def _build_ad_group_ad_operations(client, ad_group_operations):
+    """Builds new ad group ad operations.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ad_group_operations: a list of AdGroupOperation instances.
+
+    Returns: a list of AdGroupAdOperation instances.
+    """
+    return [
+        _build_ad_group_ad_operation(client, ad_group_operation) \
+        for ad_group_operation in ad_group_operations]
+
+
+def _build_ad_group_ad_operation(client, ad_group_operation):
+    """Builds a new ad group ad operation.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ad_group_operations: an AdGroupOperation instance.
+
+    Returns: an AdGroupAdOperation instance.
+    """
+    ad_group_ad_operation = client.get_type('AdGroupAdOperation', version='v3')
+    # Creates an ad group ad.
+    ad_group_ad = ad_group_ad_operation.create
+    # Creates the expanded text ad info.
+    text_ad = ad_group_ad.ad.expanded_text_ad
+    text_ad.headline_part1.value = f'Cruise to Mars #{uuid4()}'
+    text_ad.headline_part2.value = 'Best Space Cruise Line'
+    text_ad.description.value = 'Buy your tickets now!'
+    final_url = ad_group_ad.ad.final_urls.add()
+    final_url.value = 'http://www.example.com'
+    ad_group_ad.ad_group.value = ad_group_operation.create.resource_name
+    ad_group_ad.status = client.get_type('AdGroupAdStatusEnum',
+                                         version='v3').PAUSED
+
+    return ad_group_ad_operation
+
+
+def _run_mutate_job(mutate_job_service, resource_name):
+    """Runs the mutate job for executing all uploaded mutate job operations.
+
+    Args:
+        mutate_job_service: an instance of the MutateJobService message class.
+        resource_name: a str of a resource name for a mutate job.
+
+    Returns: a google.api_core.operation.Operation instance.
+    """
+    try:
+        response = mutate_job_service.run_mutate_job(resource_name)
+        print(f'Mutate job with resource name "{resource_name}" has been '
+              'executed.')
+        return response
+    except GoogleAdsException as exception:
+        _handle_google_ads_exception(exception)
+
+
+def _poll_mutate_job(operations_response, event):
+    """Polls the server until the mutate job execution finishes.
+
+    Sets the initial poll delay time and the total time to wait before time-out.
+
+    Args:
+        operations_response: a google.api_core.operation.Operation instance.
+        event: an instance of asyncio.Event to invoke once the operations have
+            completed, alerting the awaiting calling code that it can proceed.
+    """
+    loop = asyncio.get_event_loop()
+
+    def _done_callback(future):
```

Comment:
do you need future if it isn't used?

Reply:
Yes otherwise it raises an error because future is passed by the calling code inside the LRO class.